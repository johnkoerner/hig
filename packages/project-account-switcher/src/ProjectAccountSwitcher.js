import React, { Component } from "react";
import PropTypes from "prop-types";

import ProjectAccountSwitcherPresenter from "./presenters/ProjectAccountSwitcherPresenter";
import ProjectAccountSwitcherBehavior from "./behavior/ProjectAccountSwitcherBehavior";

export default class ProjectAccountSwitcher extends Component {
  static propTypes = {
    accounts: PropTypes.arrayOf(
      PropTypes.shape({
        id: PropTypes.string,
        image: PropTypes.string,
        label: PropTypes.string
      })
    ),
    accountTitle: PropTypes.string,
    activeAccountId: PropTypes.string,
    activeProjectId: PropTypes.string,
    projects: PropTypes.arrayOf(
      PropTypes.shape({
        id: PropTypes.string,
        image: PropTypes.string,
        label: PropTypes.string
      })
    ),
    projectTitle: PropTypes.string,
    onChange: PropTypes.func,
    onClick: PropTypes.func
  };

  state = {
    activeAccountId: this.props.activeAccountId,
    activeProjectId: this.props.activeProjectId
  };

  render() {
    const {
      accounts,
      accountTitle,
      activeAccountId,
      activeProjectId,
      projects,
      projectTitle,
      onChange,
      onClick
    } = this.props;

    return (
      <ProjectAccountSwitcherBehavior
        accounts={accounts}
        activeAccountId={activeAccountId}
        activeProjectId={activeProjectId}
        onChange={onChange}
        onClick={onClick}
        projects={projects}
      >
        {({
          activeAccountId,
          activeProjectId,
          handleAccountClick,
          handleProjectClick
        }) => (
          <ProjectAccountSwitcherPresenter
            accounts={accounts}
            accountTitle={accountTitle}
            activeAccountId={activeAccountId}
            activeProjectId={activeProjectId}
            onAccountClick={handleAccountClick}
            onProjectClick={handleProjectClick}
            projects={projects}
            projectTitle={projectTitle}
          />
        )}
      </ProjectAccountSwitcherBehavior>
    );
  }
}
