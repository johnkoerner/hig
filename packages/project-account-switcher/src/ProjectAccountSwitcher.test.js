import React from "react";
import renderer from "react-test-renderer";
import ProjectAccountSwitcher from "./ProjectAccountSwitcher";

describe("project-account-switcher/ProjectAccountSwitcher", () => {
  describe("integration", () => {
    it("renders correctly", () => {
      const tree = renderer
        .create(
          <ProjectAccountSwitcher
            accounts={[
              { id: "1", label: "Account 1" },
              { id: "2", label: "Account 2" }
            ]}
            accountTitle={"Accounts"}
            activeAccountId={"1"}
            activeProjectId={"2"}
            projects={[
              { id: "1", label: "Project 1" },
              { id: "2", label: "Project 2" }
            ]}
            projectTitle={"Projects"}
          />
        )
        .toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
});
