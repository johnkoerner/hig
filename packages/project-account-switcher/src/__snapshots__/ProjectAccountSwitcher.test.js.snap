// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`project-account-switcher/ProjectAccountSwitcher integration renders correctly 1`] = `
<div
  className="project-account-switcher"
>
  <div
    className="hig__flyout-v1 hig__flyout-v1--exited hig__flyout-v1--top-right"
  >
    <div
      className="hig__flyout-v1__action"
    >
      <div
        className="project-account-switcher__target"
        onClick={[Function]}
      >
        <div
          className="project-account-switcher__item"
        >
          <span
            className="project-account-switcher__item__image-wrapper"
          >
            <img
              className="project-account-switcher__item__image"
            />
            <span
              className="project-account-switcher__item__image-placeholder"
              dangerouslySetInnerHTML={
                Object {
                  "__html": "A1/P2",
                }
              }
            />
          </span>
          <span
            className="project-account-switcher__item__label"
            dangerouslySetInnerHTML={
              Object {
                "__html": "Account 1 / Project 2",
              }
            }
          />
        </div>
        <span
          className="project-account-switcher__target__caret"
        >
          <div
            className="hig__icon hig__icon--24-size"
            dangerouslySetInnerHTML={
              Object {
                "__html": "<svg width=\\"10\\" height=\\"6\\" viewBox=\\"0 0 10 6\\" xmlns=\\"http://www.w3.org/2000/svg\\" role=\\"img\\" title=\\"caret-24\\"><polyline transform=\\"rotate(45 9.828 3)\\" points=\\"8 2 8 8 2 8\\" stroke=\\"#2E4258\\" stroke-width=\\"1.15\\" fill=\\"none\\" fill-rule=\\"evenodd\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\"/></svg>",
              }
            }
          />
        </span>
      </div>
    </div>
    <div
      className="hig__flyout-v1__container"
      style={
        Object {
          "left": undefined,
          "top": undefined,
        }
      }
    >
      <div
        className="hig__flyout-v1__chevron"
      />
      <div
        className="hig__flyout-v1__panel"
        onScroll={undefined}
        style={
          Object {
            "maxHeight": undefined,
          }
        }
      >
        <div
          className="project-account-switcher__lists"
        >
          <div
            className="project-account-switcher__list"
          >
            <div
              className="project-account-switcher__list__title"
              dangerouslySetInnerHTML={
                Object {
                  "__html": "Accounts",
                }
              }
            />
            <div
              className="project-account-switcher__item project-account-switcher__item--account project-account-switcher__item--active"
              data-account-id="1"
              onClick={[Function]}
            >
              <span
                className="project-account-switcher__item__image-wrapper"
              >
                <span
                  className="project-account-switcher__item__image-placeholder"
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "A1",
                    }
                  }
                  data-account-id="1"
                />
              </span>
              <span
                className="project-account-switcher__item__label"
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "Account 1",
                  }
                }
                data-account-id="1"
              />
            </div>
            <div
              className="project-account-switcher__item project-account-switcher__item--account project-account-switcher__item--active"
              data-account-id="2"
              onClick={[Function]}
            >
              <span
                className="project-account-switcher__item__image-wrapper"
              >
                <span
                  className="project-account-switcher__item__image-placeholder"
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "A2",
                    }
                  }
                  data-account-id="2"
                />
              </span>
              <span
                className="project-account-switcher__item__label"
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "Account 2",
                  }
                }
                data-account-id="2"
              />
            </div>
          </div>
          <div
            className="project-account-switcher__list"
          >
            <div
              className="project-account-switcher__list__title"
              dangerouslySetInnerHTML={
                Object {
                  "__html": "Projects",
                }
              }
            />
            <div
              className="project-account-switcher__item project-account-switcher__item--project project-account-switcher__item--active"
              data-project-id="1"
              onClick={[Function]}
            >
              <span
                className="project-account-switcher__item__image-wrapper"
              >
                <span
                  className="project-account-switcher__item__image-placeholder"
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "P1",
                    }
                  }
                  data-project-id="1"
                />
              </span>
              <span
                className="project-account-switcher__item__label"
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "Project 1",
                  }
                }
                data-project-id="1"
              />
            </div>
            <div
              className="project-account-switcher__item project-account-switcher__item--project project-account-switcher__item--active"
              data-project-id="2"
              onClick={[Function]}
            >
              <span
                className="project-account-switcher__item__image-wrapper"
              >
                <span
                  className="project-account-switcher__item__image-placeholder"
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "P2",
                    }
                  }
                  data-project-id="2"
                />
              </span>
              <span
                className="project-account-switcher__item__label"
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "Project 2",
                  }
                }
                data-project-id="2"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
